File: truckdoc-client-m2/truckdoc-client-updater/src/main/java/com/sanda/truckdoc/client/updater/network/Backend.java
package com.sanda.truckdoc.client.updater.network;

import retrofit.http.GET;
import retrofit.http.Query;
import io.reactivex.rxjava3.core.Observable;

/**
 * Created by astra on 07.06.2015.
 */
public interface Backend {

    @GET("/check_update")
    Observable<UpdateResponse> checkUpdates(@Query("applicationId") String applicationId,
                                            @Query("applicationVersion") String versionName,
                                            @Query("androidVersion") int androidVersion,
                                            @Query("applicationCode") int versionCode,
                                            @Query("updaterVersionCode") int updaterVersionCode,
                                            @Query("deviceId") String deviceId,
                                            @Query("androidId") String androidId,
                                            @Query("deviceUuid") String deviceUuid);
}

File: truckdoc-client-m2/truckdoc-client-updater/src/main/java/com/sanda/truckdoc/client/updater/network/Backend.kt
package com.sanda.truckdoc.client.updater.network

import okhttp3.ResponseBody
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Url

/**
 * Created by astra on 07.06.2015.
 */
interface Backend {
    @GET("/check_update")
    suspend fun checkUpdates(@Query("applicationId") applicationId: String?,
                             @Query("applicationVersion") versionName: String?,
                             @Query("androidVersion") androidVersion: Int,
                             @Query("applicationCode") applicationCode: Int,
                             @Query("updaterVersionCode") updaterVersionCode: Int,
                             @Query("deviceId") deviceId: String?,
                             @Query("androidId") androidId: String?,
                             @Query("deviceUuid") deviceUuid: String?): UpdateResponse

    @GET
    suspend fun downloadFile(@Url url: String?): ResponseBody
}
